bundles/org.openhab.binding.automower/src/main/java/org/openhab/binding/automower/internal/things/AutomowerHandler.java

```diff
@@ -282,57 +938,406 @@ private void updateChannelState(@Nullable Mower mower) {
+            updateState(CHANNEL_STATISTIC_DOWN_TIME,
+                    new QuantityType<>(mower.getAttributes().getStatistics().getDownTime(), Units.SECOND));
+            updateState(CHANNEL_STATISTIC_NUMBER_OF_CHARGING_CYCLES,
+                    new DecimalType(mower.getAttributes().getStatistics().getNumberOfChargingCycles()));
+            updateState(CHANNEL_STATISTIC_NUMBER_OF_COLLISIONS,
+                    new DecimalType(mower.getAttributes().getStatistics().getNumberOfCollisions()));
+            updateState(CHANNEL_STATISTIC_TOTAL_CHARGING_TIME,
+                    new QuantityType<>(mower.getAttributes().getStatistics().getTotalChargingTime(), Units.SECOND));
+            updateState(CHANNEL_STATISTIC_TOTAL_CUTTING_TIME,
+                    new QuantityType<>(mower.getAttributes().getStatistics().getTotalCuttingTime(), Units.SECOND));
+            updateState(CHANNEL_STATISTIC_TOTAL_DRIVE_DISTANCE,
+                    new QuantityType<>(mower.getAttributes().getStatistics().getTotalDriveDistance(), SIUnits.METRE));
+            updateState(CHANNEL_STATISTIC_TOTAL_RUNNING_TIME,
+                    new QuantityType<>(mower.getAttributes().getStatistics().getTotalRunningTime(), Units.SECOND));
+            updateState(CHANNEL_STATISTIC_TOTAL_SEARCHING_TIME,
+                    new QuantityType<>(mower.getAttributes().getStatistics().getTotalSearchingTime(), Units.SECOND));
+
+            if (mower.getAttributes().getStatistics().getTotalRunningTime() != 0) {
+                updateState(CHANNEL_STATISTIC_TOTAL_CUTTING_PERCENT,
+                        new QuantityType<>(
+                                (float) mower.getAttributes().getStatistics().getTotalCuttingTime()
+                                        / (float) mower.getAttributes().getStatistics().getTotalRunningTime() * 100.0,
+                                Units.PERCENT));
+                updateState(CHANNEL_STATISTIC_TOTAL_SEARCHING_PERCENT,
+                        new QuantityType<>(
+                                (float) mower.getAttributes().getStatistics().getTotalSearchingTime()
+                                        / (float) mower.getAttributes().getStatistics().getTotalRunningTime() * 100.0,
+                                Units.PERCENT));
             } else {
-                updateState(CHANNEL_PLANNER_NEXT_START, new DateTimeType(Instant.ofEpochMilli(nextStartTimestamp)));
+                updateState(CHANNEL_STATISTIC_TOTAL_CUTTING_PERCENT, new QuantityType<>(0, Units.PERCENT));
+                updateState(CHANNEL_STATISTIC_TOTAL_SEARCHING_PERCENT, new QuantityType<>(0, Units.PERCENT));
+            }
+            updateState(CHANNEL_STATISTIC_UP_TIME,
+                    new QuantityType<>(mower.getAttributes().getStatistics().getUpTime(), Units.SECOND));
+
+            if (capabilities.hasPosition()) {
+                // if (mower.getAttributes().getLastPosition() != null) {
```

Comment:
Can this comment be removed?

Reply:
yes, as the original code was replaced by the line above.